### PR TITLE
Wall Refactor-replace array of BoardInfo by ForumBoardInfo and ForumPostInfo

### PR DESCRIPTION
**Pull request by sqreek**
Refactor:
Please watch careful at [change mcache scope](https://github.com/Wikia/app/compare/useBoardInfo_instead_array?w=1#diff-2dc9944a49a5e29752e30701ab7be9beR104)

And what do you thing about [this aproach for constructor](https://github.com/Wikia/app/compare/useBoardInfo_instead_array?w=1#diff-75997f1216a043f522c173491ae8623bR10)

also add `w=1` because whitespace refactor

notice this is PR to Wall_refactor branch, cause there is a little more changes in queue

@Wikia/sustaining-team

### CODE REVIEW

**Comment by sqreek** [comment]
tests are on red because when this commit was pushed there was a problem with jenkins executor.
I retry tests, but it not update a status.
[Here](http://jenkins.wikia-prod:8080/job/unitTests_php_branches/28116/console) is passing build from this branch

**Comment by kamilkoterba** [comment]
LGTM

File: extensions/wikia/Forum/ForumBoard.class.php [hunk 1]
```diff
@@ -17,60 +17,15 @@ static public function getEmpty() {
...
-						'username' => $username,
-						'userprofile' => $userprofile,
-						'timestamp' => $revision->getTimestamp()
-					];
-				}
-			}
-			$this->wg->Memc->set( $memKey, $info, 60 * 60 * 12 );
-		}
-
+		$info = $this->wg->Memc->get( $memKey );
```
Comment (macbre) [inline extensions/wikia/Forum/ForumBoard.class.php:26]: Invalidate the memcache key by adding a suffix

Comment (macbre) [inline extensions/wikia/Forum/ForumBoard.class.php:26]: Introduce a short helper method for generating memcache key

**Pull request merged**